mlkem/poly.c

```diff
@@ -58,17 +54,18 @@ __CPROVER_ensures(__CPROVER_return_value < KYBER_Q)
  * Arguments: - u: Unsigned canonical modulus modulo q
  *                 to be compressed.
  ************************************************************/
-uint32_t scalar_compress_q_32(int32_t u)
-/* INDENT-OFF */
-__CPROVER_requires(0 <= u && u < KYBER_Q)
-__CPROVER_ensures(__CPROVER_return_value < 32)
-//__CPROVER_ensures(__CPROVER_return_value == (((uint32_t) u * 32 + KYBER_Q / 2) / KYBER_Q) % 32)
-/* INDENT-ON */
-{
+uint32_t scalar_compress_q_32(int32_t u) {
     uint32_t d0 = (uint32_t) u;
     d0 <<= 5;
     d0 +=  1664;
+
+    /* This multiply will exceed UINT32_MAX and wrap around */
+    /* for large values of u. This is expected and required */
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
```

Comment:
I would prefer to define those exceptions at the harness level to keep the code readable.